pkg/controller/clustersync/clustersync_controller_test.go

```diff
@@ -1860,6 +1864,51 @@ func TestReconcileClusterSync_UpsertResourcesToDelete(t *testing.T) {
 	assert.Empty(t, updatedClusterSync.Status.SyncSets[0].ResourcesToDelete, "expected no resources to delete")
 }
 
+func TestReconcileClusterSync_UpsertToSyncResourceApplyMode(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	scheme := newScheme()
+	resourcesToApply := testConfigMap("dest-namespace", "dest-name")
+	syncSet := testsyncset.FullBuilder(testNamespace, "test-syncset", scheme).Build(
+		testsyncset.ForClusterDeployments(testCDName),
+		// ResourceApplyMode is Sync
+		testsyncset.WithApplyMode(hivev1.SyncResourceApplyMode),
+		testsyncset.WithGeneration(2),
+		testsyncset.WithResources(resourcesToApply),
+	)
+	// Existing SyncStatus has no ResourcesToDelete
+	existingSyncStatus := buildSyncStatus("test-syncset",
+		withTransitionInThePast(),
+		withFirstSuccessTimeInThePast(),
+	)
+	clusterSync := clusterSyncBuilder(scheme).Build(
+		testcs.WithSyncSetStatus(existingSyncStatus),
+	)
+	syncLease := buildSyncLease(time.Now().Add(-time.Hour))
+	rt := newReconcileTest(t, mockCtrl, scheme, cdBuilder(scheme).Build(), syncSet, clusterSync, syncLease)
+	rt.mockResourceHelper.EXPECT().Apply(newApplyMatcher(resourcesToApply)).Return(resource.CreatedApplyResult, nil)
+	rt.expectedSyncSetStatuses = []hiveintv1alpha1.SyncStatus{
+		buildSyncStatus("test-syncset",
+			withFirstSuccessTimeInThePast(),
+			withObservedGeneration(2),
+			// Expected SyncStatus has ResourcesToDelete
+			withResourcesToDelete(
+				testConfigMapRef("dest-namespace", "dest-name"),
+			),
+		),
+	}
+	rt.expectUnchangedLeaseRenewTime = true
+	rt.run(t)
+
+	updatedClusterSync := &hiveintv1alpha1.ClusterSync{}
+	err := rt.c.Get(context.TODO(), client.ObjectKey{Name: testCDName, Namespace: testNamespace}, updatedClusterSync)
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
+		return
+	}
+	t.Logf("%v", updatedClusterSync.Status.SyncSets[0].ResourcesToDelete)
```

Comment:
This is not checking anything? Should it be?

Reply:
@staebler Removed and rebased